revoke/revoke.go

```diff
@@ -142,6 +158,31 @@ func certIsRevokedCRL(cert *x509.Certificate, url string) (revoked, ok bool) {
 		}
 	}
 
+	u, err := neturl.Parse(url)
+	if err != nil {
+		log.Warningf("failed to parse CRL url: %v", err)
+		return false, false
+	}
+	if u.Scheme == "" && shouldFetchCRL {
```

Comment:
If you have a file:// URL, wouldn't the `Path` field work for the file name? RFC 5280 has a [number of options](https://tools.ietf.org/html/rfc5280#section-4.2.1.13) for CRL distribution points, and the one that seems most relevant is a URI (other options include directory names, e.g. for LDAP, or `nameRelativeToIssuer`). It seems then that the file scheme is the most appropriate. If it's a local file, wouldn't it make sense to put it in CRLSet as a URI, not a plain path?